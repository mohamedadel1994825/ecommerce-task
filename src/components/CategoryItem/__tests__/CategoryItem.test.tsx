import { cleanup, render } from "@testing-library/react-native";
import React from "react";

import { CategoryItem } from "../index";

describe("CategoryItem", () => {
  afterEach(cleanup);

  const item = {
    image: "image-name",
    category: "Category Name",
  };

  it("should match snapshot", () => {
    const { toJSON } = render(
      <CategoryItem item={item} isSelected={true} onPress={() => {}} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders category text correctly", () => {
    const { getByText } = render(
      <CategoryItem item={item} isSelected={true} onPress={() => {}} />
    );
    expect(getByText(item.category)).toBeTruthy();
  });
});
