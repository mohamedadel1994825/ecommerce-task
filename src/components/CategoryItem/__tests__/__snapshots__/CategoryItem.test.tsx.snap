// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "opacity": 1,
    }
  }
  testID="category-item"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#0040ff",
          "fontSize": 12,
        },
        {
          "fontSize": 14,
        },
        {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ?
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 13,
          "height": 25,
          "marginTop": 10,
        },
        {
          "color": "#0040ff",
        },
      ]
    }
  >
    Category Name
  </Text>
  <View
    style={
      [
        {
          "borderBottomColor": "#0040ff",
          "borderBottomWidth": 2,
          "marginTop": 1,
          "width": "100%",
        },
        {
          "backgroundColor": "#0040ff",
        },
      ]
    }
  />
</View>
`;
